### PR TITLE
multi: modularise subserver handling 

### PR DESCRIPTION
**Pull request by ellemouton**
This PR is the first couple of commits of https://github.com/lightninglabs/lightning-terminal/pull/537 without the disabling logic.

### CODE REVIEW

File: subservers/manager.go [hunk 1]
```diff
@@ -259,3 +288,22 @@ func dialBackend(name, dialAddr, tlsCertPath string) (*grpc.ClientConn, error) {
 	}
 	return cc, nil
 }
+
+// readMacaroon tries to read the macaroon file at the specified path and create
+// gRPC dial options from it.
```
Comment (positiveblue) [inline subservers/manager.go:323]: Does `...and create gRPC dial options from it.` apply here?

File: terminal.go [hunk 2]
```diff
@@ -494,7 +491,10 @@ func (g *LightningTerminal) start() error {
 
 	// Initialise any connections to sub-servers that we are running in
 	// remote mode.
-	g.subServerMgr.ConnectRemoteSubServers()
+	if err := g.subServerMgr.ConnectRemoteSubServers(); err != nil {
```
Comment (positiveblue) [inline terminal.go:494]: :pray:

**Review by positiveblue — APPROVED** [review]
These changes make it so much cleaner. Thank you @ellemouton 

Note: i did not test the integrate/remote combination that panic in #537

File: subservers/manager.go (hunk 1 above)
Comment (ellemouton) [reply subservers/manager.go:323]: good catch!

Comment (ellemouton) [reply subservers/manager.go]: fixed

**Comment by ellemouton** [comment]
Thanks @positiveblue! 

> Note: i did not test the integrate/remote combination that panic in https://github.com/lightninglabs/lightning-terminal/pull/537

No worries - that panic is only (it seems) related to the situation where one sub-server is disabled. Which is not the case here.

**Comment by ellemouton** [comment]
ah - ok. I _thiiiiink_ I may have found something that would cause the panic. Will push up a change later today that will hopefully fix it 🤞

**Comment by ellemouton** [comment]
note: i have added one more commit on the end which I think will fix the panic that was noted by Jamal 🤞

**Comment by ellemouton** [comment]
one more commit with the same nature

File: perms/manager.go [hunk 3]
```diff
@@ -1,25 +1,18 @@
...
 	"gopkg.in/macaroon-bakery.v2/bakery"
 )
 
 const (
-	poolPerms    string = "pool"
-	loopPerms    string = "loop"
-	faradayPerms string = "faraday"
-	litPerms     string = "lit"
-	lndPerms     string = "lnd"
+	litPerms string = "lit"
```
Comment (guggero) [inline perms/manager.go:14]: Nice! A little bit less duplication :tada:

**Review by guggero — APPROVED** [review]
Very nice, LGTM :tada:

**Pull request merged**